src/models/pacman.c

```diff
@@ -81,18 +81,23 @@ void drawPacman(int count)
     }
 
     // Get new pacman position in grid
-    struct Position newPacmanGridPos = uiPosToGridPos(pacmanPosCopy);
+    struct Position newPacmanGridPos = getUiPosToGridPos(pacmanPosCopy);
 
     if (!arePositionEquals(pacmanGridPos, newPacmanGridPos))
     {
-
-        if (isObstacle(newPacmanGridPos))
+        // If pacman, just blit him at without updating his position
+        if (getMazeElementAt(newPacmanGridPos) == WALL)
         {
-            // If pacman, just blit him at without updating his position
             pacmanBlit(newPacman);
             return;
         }
 
+        if(getMazeElementAt(newPacmanGridPos) == SMALL_COIN || getMazeElementAt(newPacmanGridPos) == BIG_COIN)
+        {
```

Comment:
This should be done in the onPacmanGridMove()

This function is called only when the pacman enter in a new emplacement, not at each frame :)

Reply:
1dfc496ccd733de0f96c3773e6b4dab48016c137